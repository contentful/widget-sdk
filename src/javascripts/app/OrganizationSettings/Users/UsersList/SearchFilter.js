/* eslint "rulesdir/restrict-inline-styles": "warn" */
import React from 'react';
import PropTypes from 'prop-types';

export default class SearchFilter extends React.Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    filter: PropTypes.shape({
      key: PropTypes.string.isRequired,
      value: PropTypes.any,
      operator: PropTypes.function
    }).isRequired,
    options: PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        value: PropTypes.any
      })
    ).isRequired,
    onChange: PropTypes.func.isRequired
  };

  handleChange = ({ target: { value } }) => {
    const { onChange, filter, id } = this.props;

    onChange({ ...filter, id, value });
  };

  getSelectWidth() {
    const { filter, options } = this.props;
    const selected = options.find(option => option.value === filter.value);
    return `calc(${selected.label.length}ch + 40px)`;
  }

  render() {
    const { label, filter, options } = this.props;
    const classNames = `
      search__filter-pill
      users-search__filter-pill
      ${filter.value ? 'search__filter-pill--active' : ''}
    `;
    return (
      <div className={classNames} data-test-id="search-filter">
        <div
          className="search__filter-pill-label users-search__filter-pill-label"
          data-test-id="search-filter.label">
          {label}
        </div>
        <div className="search__select-value users-search__select-value">
          <select
            className="search__select users-search__select"
            value={filter.value}
            onChange={this.handleChange}
            data-test-id="search-filter.options"
            style={{ width: this.getSelectWidth() }}>
            {options.map(option => (
              <option key={option.value} value={option.value}>
                {option.label}
              </option>
            ))}
          </select>
        </div>
      </div>
    );
  }
}
