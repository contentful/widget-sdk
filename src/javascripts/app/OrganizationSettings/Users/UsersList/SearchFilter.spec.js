import React from 'react';
import { mount } from 'enzyme';
import SearchFilter from './SearchFilter.es6';

describe('SearchFilter', () => {
  const options = [
    { label: 'Any', value: '' },
    { label: 'Papaya', value: 'papaya' },
    { label: 'Banana', value: 'banana' },
    { label: 'Kiwi', value: 'kiwi' }
  ];

  let onChangeCb, component;

  beforeEach(() => {
    onChangeCb = jest.fn();
    component = mount(
      <SearchFilter
        id="fruit"
        label="Fruit"
        filter={{
          key: 'fruit',
          value: 'papaya',
          operator: null
        }}
        options={options}
        onChange={onChangeCb}
      />
    );
  });

  it('renders the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('accepts empty values', () => {
    component.setProps({
      filters: {
        key: 'foo',
        value: null
      }
    });
    expect(component).toMatchSnapshot();
  });

  it('calls the onChange callback with the updated filter', () => {
    const select = component.find('select');
    select.simulate('change', { target: { value: '' } });
    expect(onChangeCb).toHaveBeenCalledWith({
      id: 'fruit',
      key: 'fruit',
      operator: null,
      value: ''
    });
  });
});
