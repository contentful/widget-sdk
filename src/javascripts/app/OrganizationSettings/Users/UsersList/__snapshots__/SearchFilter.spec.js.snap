// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter accepts empty values 1`] = `
<SearchFilter
  filter={
    Object {
      "key": "fruit",
      "operator": null,
      "value": "papaya",
    }
  }
  filters={
    Object {
      "key": "foo",
      "value": null,
    }
  }
  label="Fruit"
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Any",
        "value": "",
      },
      Object {
        "label": "Papaya",
        "value": "papaya",
      },
      Object {
        "label": "Banana",
        "value": "banana",
      },
      Object {
        "label": "Kiwi",
        "value": "kiwi",
      },
    ]
  }
>
  <div
    className="
      search__filter-pill
      users-search__filter-pill
      search__filter-pill--active
    "
  >
    <div
      className="search__filter-pill-label users-search__filter-pill-label"
    >
      Fruit
    </div>
    <div
      className="search__select-value users-search__select-value"
    >
      <select
        className="search__select users-search__select"
        onChange={[Function]}
        style={
          Object {
            "width": "calc(6ch + 40px)",
          }
        }
        value="papaya"
      >
        <option
          key=""
          value=""
        >
          Any
        </option>
        <option
          key="papaya"
          value="papaya"
        >
          Papaya
        </option>
        <option
          key="banana"
          value="banana"
        >
          Banana
        </option>
        <option
          key="kiwi"
          value="kiwi"
        >
          Kiwi
        </option>
      </select>
    </div>
  </div>
</SearchFilter>
`;

exports[`SearchFilter renders the component 1`] = `
<SearchFilter
  filter={
    Object {
      "key": "fruit",
      "operator": null,
      "value": "papaya",
    }
  }
  label="Fruit"
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Any",
        "value": "",
      },
      Object {
        "label": "Papaya",
        "value": "papaya",
      },
      Object {
        "label": "Banana",
        "value": "banana",
      },
      Object {
        "label": "Kiwi",
        "value": "kiwi",
      },
    ]
  }
>
  <div
    className="
      search__filter-pill
      users-search__filter-pill
      search__filter-pill--active
    "
  >
    <div
      className="search__filter-pill-label users-search__filter-pill-label"
    >
      Fruit
    </div>
    <div
      className="search__select-value users-search__select-value"
    >
      <select
        className="search__select users-search__select"
        onChange={[Function]}
        style={
          Object {
            "width": "calc(6ch + 40px)",
          }
        }
        value="papaya"
      >
        <option
          key=""
          value=""
        >
          Any
        </option>
        <option
          key="papaya"
          value="papaya"
        >
          Papaya
        </option>
        <option
          key="banana"
          value="banana"
        >
          Banana
        </option>
        <option
          key="kiwi"
          value="kiwi"
        >
          Kiwi
        </option>
      </select>
    </div>
  </div>
</SearchFilter>
`;
